src/codec/framed_write.rs

```diff
@@ -193,12 +193,12 @@ where
                 match self.next {
                     Some(Next::Data(ref mut frame)) => {
                         tracing::trace!(queued_data_frame = true);
-                        let mut buf = (&mut self.buf).chain(frame.payload_mut());
-                        ready!(Pin::new(&mut self.inner).poll_write_buf(cx, &mut buf))?;
+                        let mut buf = (&mut self.buf).chain(frame.payload_mut()).to_bytes();
+                        ready!(Pin::new(&mut self.inner).poll_write(cx, &mut buf))?;
                     }
                     _ => {
                         tracing::trace!(queued_data_frame = false);
-                        ready!(Pin::new(&mut self.inner).poll_write_buf(cx, &mut self.buf))?;
+                        ready!(Pin::new(&mut self.inner).poll_write(cx, &mut self.buf.to_bytes()))?;
```

Comment:
Can we use https://docs.rs/bytes/0.5.6/bytes/buf/trait.Buf.html#tymethod.bytes here instead of `to_bytes`?

Reply:
We also need to advance the buffer position here since that is what `poll_write_buf` did.